NBitcoin/Crypto/NativeDigests/ManagedSha256Digest.cs

```diff
@@ -0,0 +1,63 @@
+﻿using NBitcoin.BouncyCastle.Crypto;
+using System;
+
+namespace NBitcoin.Crypto.NativeDigests
+{
+#if !NETSTANDARD1X && !NONATIVEHASH
+
+	using System.Security.Cryptography;
+
+	/// <summary>
+	/// A wrapper around the native SHA256, implements BouncyCastle's IDigest interface in order
+	/// to be compatible with BouncyCastle's HMac implementation.
+	/// </summary>
+	internal class ManagedSha256Digest : IDigest, IDisposable
+	{
+		private const int DigestLength = 32;
+		SHA256Managed nativeSha256 = null;
+		byte[] input;
+
+		public ManagedSha256Digest()
+		{
+			Reset();
+		}
+
+		public string AlgorithmName => "SHA-256";
+
+		public void BlockUpdate(byte[] input, int inOff, int length)
+		{
+			this.input = new byte[length];
+			Array.Copy(input, inOff, this.input, 0, length);
+		}
+
+		public int DoFinal(byte[] output, int outOff)
+		{
+			var hash = nativeSha256.ComputeHash(input, 0, input.Length);
```

Comment:
Can you instead rely on NBitcoin.Secp256k1.SHA256 class which properly implement DoFinal and BlockUpdate and is widely tested and memory efficient?

Reply:
Right, I'll take care of it tonight.

Reply:
@NicolasDorier I found the NBitcoin.Secpk1.SHA class, but it doesn't look like it implements the IDigest interface, nor the BlockUpdate/DoFinal methods.

Reply:
I am not saying to use it instead, but to use it rather than that `nativeSha256`

Reply:
Hello @NicolasDorier, I understand now. There is no NBitcoin.Secp512k1.SHA512 equivalent in the Secpk1 project though (to go along with the SHA256). Besides, wouldn't it be great if NBitcoin could rely on the framework's builtins? 

It's not clear to me if DoFinal() and BlockUpdate() are BouncyCastle-specific interface methods, or if they are a more wider known concept. I'm not sure how to implement them correctly. I was hoping someone here would review it, and fill in the gaps for me.